File: services/assessment-service/src/test/java/com/visor/school/assessment/model/AssessmentTypeTest.java
package com.visor.school.assessment.model;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class AssessmentTypeTest {

    @Test
    void shouldHaveAllRequiredAssessmentTypes() {
        AssessmentType[] types = AssessmentType.values();

        assertTrue(java.util.Arrays.asList(types).contains(AssessmentType.TEST));
        assertTrue(java.util.Arrays.asList(types).contains(AssessmentType.QUIZ));
        assertTrue(java.util.Arrays.asList(types).contains(AssessmentType.ASSIGNMENT));
        assertTrue(java.util.Arrays.asList(types).contains(AssessmentType.PROJECT));
        assertTrue(java.util.Arrays.asList(types).contains(AssessmentType.EXAM));
        assertTrue(java.util.Arrays.asList(types).contains(AssessmentType.FINAL_EXAM));
        assertEquals(6, types.length);
    }

    @Test
    void shouldHaveCorrectTypeNames() {
        assertEquals("TEST", AssessmentType.TEST.name());
        assertEquals("QUIZ", AssessmentType.QUIZ.name());
        assertEquals("ASSIGNMENT", AssessmentType.ASSIGNMENT.name());
        assertEquals("PROJECT", AssessmentType.PROJECT.name());
        assertEquals("EXAM", AssessmentType.EXAM.name());
        assertEquals("FINAL_EXAM", AssessmentType.FINAL_EXAM.name());
    }
}

File: services/academic-assessment-service/src/test/kotlin/com/visor/school/assessment/model/AssessmentTypeTest.kt
package com.visor.school.assessment.model

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

class AssessmentTypeTest {

    @Test
    fun `should have all required assessment types`() {
        val types = AssessmentType.values()

        assertTrue(types.contains(AssessmentType.TEST))
        assertTrue(types.contains(AssessmentType.QUIZ))
        assertTrue(types.contains(AssessmentType.ASSIGNMENT))
        assertTrue(types.contains(AssessmentType.PROJECT))
        assertTrue(types.contains(AssessmentType.EXAM))
        assertTrue(types.contains(AssessmentType.FINAL_EXAM))
        assertEquals(6, types.size)
    }

    @Test
    fun `should have correct type names`() {
        assertEquals("TEST", AssessmentType.TEST.name)
        assertEquals("QUIZ", AssessmentType.QUIZ.name)
        assertEquals("ASSIGNMENT", AssessmentType.ASSIGNMENT.name)
        assertEquals("PROJECT", AssessmentType.PROJECT.name)
        assertEquals("EXAM", AssessmentType.EXAM.name)
        assertEquals("FINAL_EXAM", AssessmentType.FINAL_EXAM.name)
    }
}

